File: mobile/fetch_logs.py
import urllib.request
import json
import subprocess

# Get URL from EAS
result = subprocess.run(['npx', 'eas', 'build:view', 'f6353991-4ca8-4382-8268-43cf867b05fe', '--json'], capture_output=True, text=True, shell=True)
data = json.loads(result.stdout)
url = data['logFiles'][0]

print("Downloading logs from:", url)
req = urllib.request.Request(url)
with urllib.request.urlopen(req) as response:
    content = response.read()

print("First 50 bytes of content:", content[:50])

File: mobile/fetch_logs.mjs
import { execSync } from 'child_process';
import fs from 'fs';

try {
  const result = execSync('npx eas build:view 1933780f-f71e-4840-a335-18877d4f61c9 --json', { encoding: 'utf-8' });
  const data = JSON.parse(result);
  const url = data.logFiles[0];
  
  const response = await fetch(url);
  const text = await response.text();
  
  const lines = text.split('\n');
  const errors = [];
  
  for (const line of lines) {
    if (!line.trim()) continue;
    try {
      const j = JSON.parse(line);
      if (j.phase === 'RUN_GRADLEW') {
        errors.push(j.msg || '');
      }
    } catch(e) {}
  }
  
  console.log(errors.slice(-300).join('\n'));
} catch (e) {
  console.error(e);
}
